Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -275,35 +275,46 @@ private FunctionEnvelope CreateFunctionConfig(string configContent, string funct
             };
         }
 
-        private string GetFunctionScriptPath(string functionName, JObject functionConfig)
-        {
-            var functionPath = GetFunctionPath(functionName);
-            var functionFiles = FileSystemHelpers.GetFiles(functionPath, "*.*", SearchOption.TopDirectoryOnly)
-                .Where(p => Path.GetFileName(p).ToLowerInvariant() != "function.json").ToArray();
-
-            return DeterminePrimaryScriptFile(functionConfig, functionFiles);
-        }
-
         // Logic for this function is copied from here:
         // https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs
         // These two implementations must stay in sync!
-        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
+
+        /// <summary>
+        /// Determines which script should be considered the "primary" entry point script.
+        /// </summary>
+        /// <exception cref="ConfigurationErrorsException">Thrown if the function metadata points to an invalid script file, or no script files are present.</exception>
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string scriptDirectory)
         {
-            if (functionFiles.Length == 1)
+            // First see if there is an explicit primary file indicated
+            // in config. If so use that.
+            string functionPrimary = null;
+            string scriptFile = (string)functionConfig["scriptFile"];
+
+            if (!string.IsNullOrEmpty(scriptFile))
             {
-                // if there is only a single file, that file is primary
-                return functionFiles[0];
+                string scriptPath = Path.Combine(scriptDirectory, scriptFile);
```

Comment:
Note that in some cases, the script file will be outside the folder (e.g. `../otherfolder/foo.js`). e.g. this will happen in scenarios like @christopheranderson's funcpack.